### PR TITLE
[Doc] [KubeRay] Add GKE ingress doc

### PR DESCRIPTION
**Pull request by architkulkarni**
## Why are these changes needed?
Adds a doc section for setting up Ingress on GKE.  At the end of the doc, the user will be able to view the Ray dashboard UI via the Ingress.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/cluster/kubernetes/k8s-ecosystem/ingress.md [hunk 1]
```diff
@@ -72,6 +73,70 @@ kubectl describe ingress ray-cluster-ingress
...
+
+# Step 3: Edit ray-cluster-gclb-ingress.yaml to replace the service name with the name of the head service from the RayCluster. (Output of `kubectl get svc`)
+
+# Step 4: Apply the Ingress configuration
+kubectl apply -f ray-cluster-gclb-ingress.yaml
+
+# Step 5: Check ingress created by Step 4.
+kubectl describe ingress ray-cluster-ingress
+
+# Step 6: After a few minutes, GKE will allocate an external IP for the ingress. Check it using:
```
Comment (angelinalg) [inline doc/source/cluster/kubernetes/k8s-ecosystem/ingress.md:127]:
```suggestion
# Step 6: After a few minutes, GKE allocates an external IP for the ingress. Check it using:
```

**Review by angelinalg — APPROVED** [review]
Just one nit for style. Sneak peek: You can use the Vale linter to check for style. See go/vale.

File: doc/source/cluster/kubernetes/k8s-ecosystem/ingress.md [hunk 2]
```diff
@@ -72,6 +73,70 @@ kubectl describe ingress ray-cluster-ingress
...
+    kubernetes.io/ingress.class: "gce"
+spec:
+  rules:
+    - http:
+        paths:
+          - path: /
+            pathType: Prefix
+            backend:
+              service:
+                name: raycluster-mini-head-svc # Update this line with your head service in Step 3 below.
```
Comment (kevin85421) [inline doc/source/cluster/kubernetes/k8s-ecosystem/ingress.md:104]:
```suggestion
                name: raycluster-kuberay-head-svc # Update this line with your head service in Step 3 below.
```

**Review by kevin85421 — APPROVED** [review]
LGTM

**Pull request merged**